src/Microsoft.Azure.SignalR/ProductInfo.cs

```diff
@@ -1,22 +1,49 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Linq;
 using System.Reflection;
 using System.Runtime.InteropServices;
+using Microsoft.Azure.SignalR.Common;
 
 namespace Microsoft.Azure.SignalR
 {
     internal static class ProductInfo
     {
+        private const int MaxProductInfoLength = 256;
         public static string GetProductInfo()
         {
-            var assembly = Assembly.GetCallingAssembly();
+            var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
```

Comment:
Maybe just make it for Management API?

Reply:
But management SDK reuse service SDK, so there seems no way to tell which one is quering product info if we don't change api.